### PR TITLE
Domains: Disable non-owner actions and notifications

### PR DESCRIPTION
**Pull request by aidvu**
- Disable Edit Contact Info for non-owners
- Disable Transfer for non-owners
- Don't show notices for domains the user doesn't own

Needs D2177-code.

Steps:
Scenarion 1:
1. Create a blog with 2 users (user A and B)
2. Register a domain with user A
3. Login with user B
4. Go to Domain Management -> newly registered domain
5. Click on Contact Info and Privacy
6. Click on Edit Contact Info
7. You should see `Only owner is allowed to perform this action.`
8. Go back to the Domain Screen
9. Click `Transfer Domain`
10. You should see `Only owner is allowed to perform this action.`

Scenario 2:
1. Create a blog with 2 users (user A and B)
2. Register a domain with user A
3. On the domain management screen you should see a notice about an unverified domain
4. Login with user B
5. You shouldn't see the notice, as this user isn't the owner

@klimeryk @umurkontaci 

Test live: https://calypso.live/?branch=update/disable-non-owner-actions

### CODE REVIEW

File: client/my-sites/upgrades/components/domain-warnings/index.jsx [hunk 1]
```diff
@@ -52,19 +52,20 @@ export default React.createClass( {
...
-			let validRules = this.props.ruleWhiteList.map( ruleName => this[ ruleName ] );
+			const validRules = this.props.ruleWhiteList.map( ruleName => this[ ruleName ] );
 			rules = intersection( allRules, validRules ); // avoid leaking other functions
 		}
 		return rules;
 	},
 
 	getDomains() {
-		return this.props.domains || [ this.props.domain ];
+		return ( this.props.domains || [ this.props.domain ] ).filter( domain => domain.isCurrentUserOwner );
```
Comment (klimeryk) [inline client/my-sites/upgrades/components/domain-warnings/index.jsx]: Very nice! 👍

**Comment by aidvu** [comment]
@breezyskies I'd like your input on this. We haven't hidden the options for non-owners, but instead show a `Card` saying `Only owner can perform this action.`.

**Comment by breezyskies** [comment]
> We haven't hidden the options for non-owners, but instead show a Card saying Only owner can perform this action..

+1 for this (vs. hiding the options), but the messaging might need some work. cc: @ranh

**Comment by ranh** [comment]
How about:

> These settings can be changed by the user that the domain is registered to.

It's not a great line but it gets the point across I think. 

Is it correct to say "the user that the domain is registered to"? 

We can say "the user who owns the domain" or just "the domain's owner", but that seems problematic too. Owning the domain is not the same as being logged in with the account that the subscription is tied to. And generally, the idea of "owning" subscriptions may not be familiar or clear enough to most users, since it only very rarely comes up.

**Comment by aidvu** [comment]
If we don't have an idea of `ownership`, then maybe: 
> These settings can be changed by the user who registered the domain.

**Comment by ranh** [comment]
> For some reason These settings can be changed by the user that the domain is registered to. this doesn't sound right to me. Maybe ...registered with.. 😕

@aidvu It's in passive voice and it's a complicated sentence, I agree it doesn't sound great. I didn't use "the user who registered the domain" because that may not be accurate. The subscription / registration can change hands. Maybe I'm splitting hairs though. Your version does sound better.

**Comment by aidvu** [comment]
Updated copy, and made the `NonOwnerCard` a function, since it doesn't have a state.

**Comment by klimeryk** [comment]
Tested and it works great 👍 
However I have two suggestions ("Here we go again..." 😜 ):

1. `These settings can be changed by the user who registered the domain.` - how about adding `only` there and specifying which domain exactly we are talking about. Some users think that domain == site, so they might be thinking "I'm the owner of the site, I should be able to do this!", but in reality the owner of the site and the owner of the domain can be two different users, so this might help avoiding this confusion.
1. The component itself looks a bit bland - like it's missing a style or something. Maybe you could use a `Notice` instead?

All in all, it'd look something like this:
screen shot 2016-07-08 at 13 02 23
Instead of:
screen shot 2016-07-08 at 13 07 55

**Comment by ranh** [comment]
I intentionally left out "only" since I felt it adds a negative tone to the message. It's implied by the context of the message so it's not necessary to explicitly state it. Do you think the meaning is lost without it?

+1 to specifying the domain. Even though it's shown in the title above, I agree it can be helpful to clarify any confusion about what the notice is referencing.

**Comment by aidvu** [comment]
I like the notice idea! Will add the domain in there too. 🙇

**Comment by klimeryk** [comment]
> I intentionally left out "only" since I felt it adds a negative tone to the message. It's implied by the context of the message so it's not necessary to explicitly state it. Do you think the meaning is lost without it?

You're right - the version without "only" sounds more positive. Good call! :)

**Comment by aidvu** [comment]
@klimeryk updated.

**Comment by aidvu** [comment]
@klimeryk domain <stronk />!

**Comment by klimeryk** [comment]
🚢 it stronk!

**Pull request merged**